src/main/java/webserver/RequestHandler.java

```diff
@@ -2,54 +2,96 @@
 
 import java.io.*;
 import java.net.Socket;
-import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
 
+import model.User;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import util.RequestUtil;
+import util.RequestLineParser;
 
 public class RequestHandler implements Runnable {
     private static final String DEFAULT_PATH = "./src/main/resources/static";
+    private static final String SIGN_UP_URL_PATH = "/register.html";
+
     private static final Logger logger = LoggerFactory.getLogger(RequestHandler.class);
     private Socket connection;
+    private Map<String, String> httpHeaders = new HashMap<>();
 
     public RequestHandler(Socket connectionSocket) { // 소켓 타입의 인자를 받아 connection 필드에 저장
         this.connection = connectionSocket;
     }
 
     public void run() {
         // 클라이언트 연결 정보 로깅 : 클라이언트가 연결되면, IP 주소와 포트 번호를 로깅한다.
-        logger.debug("New Client Connect! Connected IP : {}, Port : {}", connection.getInetAddress(),
-                connection.getPort());
-        // 입출력 스트림 준비 : 클라이언트와의 데이터 교환을 위해 입력 스트림과 출력 스트림을 준비한다.
-            // 브라우저에서 서버쪽으로 들어오는 모든 데이터는 InputStream 에 담겨있음
-            // 서버에서 브라우저로의 응답은 OutStream 에 실어서 보내다.
+        logger.debug("New Client Connect! Connected IP : {}, Port : {}", connection.getInetAddress(), connection.getPort());
         try (InputStream in = connection.getInputStream(); OutputStream out = connection.getOutputStream()) {
-            // 일반적으로 헤더는 라인 단위로 구성된다. 라인 단위로 데이터를 읽기 위해 IntStream -> BufferedReader 로 변경한다.
             BufferedReader br = new BufferedReader(new InputStreamReader(in, "UTF-8"));
+            DataOutputStream dos = new DataOutputStream(out);
 
             // 첫 번째 라인에서 요청 URL 추츨 (/index.html)
             String line = br.readLine();
-            String url = RequestUtil.getUrl(line);
+            RequestLineParser requestLineParser = new RequestLineParser(line);
```

Comment:
```suggestion
            RequestLineParser requestLineParser = new RequestLineParser(br.readLine());
```
보통은 이렇게 합치는 걸 선호하는 편입니다.